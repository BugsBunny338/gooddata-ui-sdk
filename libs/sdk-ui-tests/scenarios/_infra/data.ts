// (C) 2020 GoodData Corporation

/*
 * In some cases, it is still valid to reference data elements by their URI - because URI is
 * essentially the primary key to identify particular entity (yes, bear's URI have some drawbacks).
 *
 * This is essential when sorting pivot table or doing color mapping for charts.
 *
 * NOTE: these values figure in test scenario insight converters; if you change this it
 *  is essential to run 'populate-ref' script to make sure insights created in the
 *  reference workspace are up-to-date.
 *
 * TODO Ideally, a map like this should instead be included in recording index and be auto-generated by the
 *  mock tooling and be part of the reference workspace; good starting task for someone interested in typescript
 *  code-gen done in mock-handling tool.
 */

export const Region = {
    EastCoast: "/gdc/md/l32xdyl4bjuzgf9kkqr2avl55gtuyjwf/obj/1086/elements?id=460488",
    WestCoast: "/gdc/md/l32xdyl4bjuzgf9kkqr2avl55gtuyjwf/obj/1086/elements?id=460489",
};

export const Product = {
    Explorer: "/gdc/md/l32xdyl4bjuzgf9kkqr2avl55gtuyjwf/obj/1054/elements?id=342345",
    WonderKid: "/gdc/md/l32xdyl4bjuzgf9kkqr2avl55gtuyjwf/obj/1054/elements?id=166497",
};
