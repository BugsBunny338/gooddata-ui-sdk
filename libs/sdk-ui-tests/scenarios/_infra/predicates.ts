// (C) 2007-2019 GoodData Corporation

import { HeaderPredicateFactory } from "@gooddata/sdk-ui";

/**
 * Predicates matching particular attribute elements
 */
export const AttributeElements = {
    Product: {
        Explorer: HeaderPredicateFactory.attributeItemNameMatch("Explorer"),
        WonderKid: HeaderPredicateFactory.attributeItemNameMatch("WonderKid"),
    },
    Region: {
        EastCoast: HeaderPredicateFactory.attributeItemNameMatch("East Coast"),
        WestCoast: HeaderPredicateFactory.attributeItemNameMatch("West Coast"),
    },
};
