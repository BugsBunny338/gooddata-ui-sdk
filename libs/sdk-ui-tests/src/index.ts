// (C) 2007-2020 GoodData Corporation

export * from "./scenarioGroup";
export * from "./scenario";
export * from "./scenarioTestInput";
