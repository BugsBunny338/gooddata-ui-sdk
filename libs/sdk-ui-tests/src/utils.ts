// (C) 2007-2019 GoodData Corporation

import cloneDeepWith = require("lodash/cloneDeepWith");
import isFunction = require("lodash/isFunction");
import isObject = require("lodash/isObject");
import omit = require("lodash/omit");

/**
 * Cleans up properties before taking their snapshot. The goal of this function is to remove any properties
 * that are not significant for the snapshot testing. For instance:
 *
 * -  completely remove the 'backend' prop
 * -  clean up functions from prepared execution stored in 'execution' prop
 *
 * @param props - props to clean up
 * @param omitKeys - keys to completely remove from the props; defaults to omiting the 'backend' prop
 * @param cleanupFuns - keys whose values are objects that should be cleaned from any props that contain functions, defaults to 'execution'
 */
export function cleanupProps(
    props: any,
    omitKeys: string[] = ["backend"],
    cleanupFuns: string[] = ["execution"],
): any {
    const propsWithoutBackend = omit(props, omitKeys);
    const cleanup = (value: any, key: string) => {
        if (cleanupFuns.includes(key) && isObject(value)) {
            Object.keys(value).forEach(key => {
                if (isFunction(value[key])) {
                    delete value[key];
                }
            });
        }
    };

    return cloneDeepWith(propsWithoutBackend, cleanup);
}
