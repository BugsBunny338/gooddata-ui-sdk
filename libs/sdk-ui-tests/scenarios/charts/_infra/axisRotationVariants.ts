// (C) 2007-2019 GoodData Corporation
import { ScenarioNameAndProps, UnboundVisProps, VisProps } from "../../../src";
import { IChartConfig } from "@gooddata/sdk-ui";
import cloneDeep = require("lodash/cloneDeep");

const ConfigVariants: Array<[string]> = [["90"], ["-90"], ["60"], ["-60"]];

function updateAxisSettings(rotation: string, config: IChartConfig = {}): IChartConfig {
    const copy = cloneDeep(config);

    if (copy.xaxis) {
        copy.xaxis.rotation = rotation;
    }

    if (copy.secondary_xaxis) {
        copy.secondary_xaxis.rotation = rotation;
    }

    return copy;
}

export function axisRotationVariants<T extends VisProps>(
    baseName: string,
    baseProps: UnboundVisProps<T>,
): Array<ScenarioNameAndProps<T>> {
    return ConfigVariants.map(([rotation]) => {
        /*
         * stories use the scenario name as story name. the minus sign messes things up and would lead
         * to duplicate stories..
         */
        return [
            `${baseName} - ${rotation.replace("-", "minus")}`,
            { ...baseProps, config: updateAxisSettings(rotation, baseProps.config) },
        ];
    });
}
