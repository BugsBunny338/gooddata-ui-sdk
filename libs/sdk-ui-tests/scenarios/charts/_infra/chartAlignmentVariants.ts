// (C) 2007-2019 GoodData Corporation
import { IChartConfig } from "@gooddata/sdk-ui-charts";
import { CustomizedScenario, UnboundVisProps, VisProps } from "../../../src";

const ConfigVariants: Array<[string, IChartConfig]> = [
    ["bottom", { chart: { verticalAlign: "bottom" } }],
    ["middle", { chart: { verticalAlign: "middle" } }],
    ["top", { chart: { verticalAlign: "top" } }],
    ["default", {}],
];

export function chartAlignmentVariants<T extends VisProps>(
    baseName: string,
    baseProps: UnboundVisProps<T>,
): Array<CustomizedScenario<T>> {
    return ConfigVariants.map(([variantName, config]) => {
        return [`${baseName} - ${variantName}`, { ...baseProps, config }];
    });
}
