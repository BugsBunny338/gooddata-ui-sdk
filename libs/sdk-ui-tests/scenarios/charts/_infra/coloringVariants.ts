// (C) 2007-2019 GoodData Corporation

import { IChartConfig } from "@gooddata/sdk-ui-charts";
import { CustomizedScenario, ScenarioTag, UnboundVisProps, VisProps } from "../../../src";
import { CustomColorPalette, CustomColors } from "../../_infra/colors";

const ConfigVariants: Array<[string, IChartConfig, ScenarioTag[]?]> = [
    ["default", {}, ["mock-no-insight"]],
    ["custom palette", { colorPalette: CustomColorPalette }],
    ["custom colors", { colors: CustomColors }, ["mock-no-insight"]],
    [
        "custom palette preferred over colors",
        {
            colorPalette: CustomColorPalette,
            colors: CustomColors,
        },
        ["mock-no-insight"],
    ],
];

export function coloringCustomizer<T extends VisProps>(
    baseName: string,
    baseProps: UnboundVisProps<T>,
    baseTags: ScenarioTag[],
): Array<CustomizedScenario<T>> {
    return ConfigVariants.map(([variantName, coloringOverlay, tags = []]) => {
        return [
            `${baseName} - ${variantName}`,
            { ...baseProps, config: { ...baseProps.config, ...coloringOverlay } },
            [...baseTags, ...tags],
        ];
    });
}
