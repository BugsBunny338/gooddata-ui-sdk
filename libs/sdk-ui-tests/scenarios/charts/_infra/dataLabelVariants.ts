// (C) 2007-2019 GoodData Corporation

import { IChartConfig } from "@gooddata/sdk-ui";
import { CustomizedScenario, UnboundVisProps, VisProps } from "../../../src";
import { GermanNumberFormat } from "../../_infra/formatting";

const ConfigVariants: Array<[string, IChartConfig]> = [
    ["default", {}],
    ["auto visibility", { dataLabels: { visible: "auto" } }],
    ["forced visible", { dataLabels: { visible: true } }],
    [
        "forced visible and german separators",
        { dataLabels: { visible: true }, separators: GermanNumberFormat },
    ],
    ["forced hidden", { dataLabels: { visible: false } }],
];

export function dataLabelCustomizer<T extends VisProps>(
    baseName: string,
    baseProps: UnboundVisProps<T>,
): Array<CustomizedScenario<T>> {
    return ConfigVariants.map(([variantName, dataLabelOverlay]) => {
        return [
            `${baseName} - ${variantName}`,
            { ...baseProps, config: { ...baseProps.config, ...dataLabelOverlay } },
        ];
    });
}
