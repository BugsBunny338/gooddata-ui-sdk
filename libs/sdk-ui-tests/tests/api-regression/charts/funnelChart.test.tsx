// (C) 2007-2019 GoodData Corporation

import { IFunnelChartProps } from "@gooddata/sdk-ui";
import funnelChartScenarios from "../../../scenarios/charts/funnelChart";
import { ScenarioTestInput } from "../../../src";
import { mountChartAndCapture } from "../../_infra/render";
import { cleanupCoreChartProps } from "../../_infra/utils";
import flatMap = require("lodash/flatMap");

describe("FunnelChart", () => {
    const Scenarios: Array<ScenarioTestInput<IFunnelChartProps>> = flatMap(funnelChartScenarios, group =>
        group.forTestTypes("api").asTestInput(),
    );

    describe.each(Scenarios)("with %s", (_desc, Component, propsFactory) => {
        const interactions = mountChartAndCapture(Component, propsFactory);

        it("should create expected execution definition", () => {
            expect(interactions.triggeredExecution).toMatchSnapshot();
        });

        it("should create expected props for core chart", () => {
            expect(interactions.passedToBaseChart).toBeDefined();
            expect(interactions.passedToBaseChart!.execution).toBeDefined();
            expect(cleanupCoreChartProps(interactions.passedToBaseChart)).toMatchSnapshot();
        });
    });
});
