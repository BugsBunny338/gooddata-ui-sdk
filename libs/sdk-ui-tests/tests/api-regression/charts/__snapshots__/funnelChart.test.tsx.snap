// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FunnelChart with  - custom colors should create expected execution definition 1`] = `
Object {
  "attributes": Array [
    Object {
      "attribute": Object {
        "displayForm": Object {
          "identifier": "label.product.id.name",
        },
        "localIdentifier": "a_label.product.id.name",
      },
    },
  ],
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "ah1EuQxwaCqs",
                },
              },
            },
            "localIdentifier": "m_ah1EuQxwaCqs",
          },
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [
        Object {
          "attribute": Object {
            "displayForm": Object {
              "identifier": "label.product.id.name",
            },
            "localIdentifier": "a_label.product.id.name",
          },
        },
      ],
      "localIdentifier": "view",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [
        "measureGroup",
      ],
    },
    Object {
      "itemIdentifiers": Array [
        "a_label.product.id.name",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "ah1EuQxwaCqs",
            },
          },
        },
        "localIdentifier": "m_ah1EuQxwaCqs",
      },
    },
  ],
  "sortBy": Array [],
  "workspace": "testWorkspace",
}
`;

exports[`FunnelChart with  - custom colors should create expected props for core chart 1`] = `
Object {
  "config": Object {
    "colors": Array [
      "rgb(33, 41, 195)",
      "rgb(18, 194, 25)",
      "rgb(243, 56, 47)",
      "rgb(240, 239, 8)",
      "rgb(131, 131, 131)",
    ],
  },
}
`;

exports[`FunnelChart with  - custom palette preferred over colors should create expected execution definition 1`] = `
Object {
  "attributes": Array [
    Object {
      "attribute": Object {
        "displayForm": Object {
          "identifier": "label.product.id.name",
        },
        "localIdentifier": "a_label.product.id.name",
      },
    },
  ],
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "ah1EuQxwaCqs",
                },
              },
            },
            "localIdentifier": "m_ah1EuQxwaCqs",
          },
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [
        Object {
          "attribute": Object {
            "displayForm": Object {
              "identifier": "label.product.id.name",
            },
            "localIdentifier": "a_label.product.id.name",
          },
        },
      ],
      "localIdentifier": "view",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [
        "measureGroup",
      ],
    },
    Object {
      "itemIdentifiers": Array [
        "a_label.product.id.name",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "ah1EuQxwaCqs",
            },
          },
        },
        "localIdentifier": "m_ah1EuQxwaCqs",
      },
    },
  ],
  "sortBy": Array [],
  "workspace": "testWorkspace",
}
`;

exports[`FunnelChart with  - custom palette preferred over colors should create expected props for core chart 1`] = `
Object {
  "config": Object {
    "colorPalette": Array [
      Object {
        "fill": Object {
          "b": 73,
          "g": 49,
          "r": 195,
        },
        "guid": "01",
      },
      Object {
        "fill": Object {
          "b": 86,
          "g": 194,
          "r": 168,
        },
        "guid": "02",
      },
      Object {
        "fill": Object {
          "b": 177,
          "g": 217,
          "r": 243,
        },
        "guid": "03",
      },
      Object {
        "fill": Object {
          "b": 121,
          "g": 153,
          "r": 194,
        },
        "guid": "04",
      },
      Object {
        "fill": Object {
          "b": 34,
          "g": 37,
          "r": 162,
        },
        "guid": "05",
      },
    ],
    "colors": Array [
      "rgb(33, 41, 195)",
      "rgb(18, 194, 25)",
      "rgb(243, 56, 47)",
      "rgb(240, 239, 8)",
      "rgb(131, 131, 131)",
    ],
  },
}
`;

exports[`FunnelChart with  - custom palette should create expected execution definition 1`] = `
Object {
  "attributes": Array [
    Object {
      "attribute": Object {
        "displayForm": Object {
          "identifier": "label.product.id.name",
        },
        "localIdentifier": "a_label.product.id.name",
      },
    },
  ],
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "ah1EuQxwaCqs",
                },
              },
            },
            "localIdentifier": "m_ah1EuQxwaCqs",
          },
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [
        Object {
          "attribute": Object {
            "displayForm": Object {
              "identifier": "label.product.id.name",
            },
            "localIdentifier": "a_label.product.id.name",
          },
        },
      ],
      "localIdentifier": "view",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [
        "measureGroup",
      ],
    },
    Object {
      "itemIdentifiers": Array [
        "a_label.product.id.name",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "ah1EuQxwaCqs",
            },
          },
        },
        "localIdentifier": "m_ah1EuQxwaCqs",
      },
    },
  ],
  "sortBy": Array [],
  "workspace": "testWorkspace",
}
`;

exports[`FunnelChart with  - custom palette should create expected props for core chart 1`] = `
Object {
  "config": Object {
    "colorPalette": Array [
      Object {
        "fill": Object {
          "b": 73,
          "g": 49,
          "r": 195,
        },
        "guid": "01",
      },
      Object {
        "fill": Object {
          "b": 86,
          "g": 194,
          "r": 168,
        },
        "guid": "02",
      },
      Object {
        "fill": Object {
          "b": 177,
          "g": 217,
          "r": 243,
        },
        "guid": "03",
      },
      Object {
        "fill": Object {
          "b": 121,
          "g": 153,
          "r": 194,
        },
        "guid": "04",
      },
      Object {
        "fill": Object {
          "b": 34,
          "g": 37,
          "r": 162,
        },
        "guid": "05",
      },
    ],
  },
}
`;

exports[`FunnelChart with  - default should create expected execution definition 1`] = `
Object {
  "attributes": Array [
    Object {
      "attribute": Object {
        "displayForm": Object {
          "identifier": "label.product.id.name",
        },
        "localIdentifier": "a_label.product.id.name",
      },
    },
  ],
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "ah1EuQxwaCqs",
                },
              },
            },
            "localIdentifier": "m_ah1EuQxwaCqs",
          },
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [
        Object {
          "attribute": Object {
            "displayForm": Object {
              "identifier": "label.product.id.name",
            },
            "localIdentifier": "a_label.product.id.name",
          },
        },
      ],
      "localIdentifier": "view",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [
        "measureGroup",
      ],
    },
    Object {
      "itemIdentifiers": Array [
        "a_label.product.id.name",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "ah1EuQxwaCqs",
            },
          },
        },
        "localIdentifier": "m_ah1EuQxwaCqs",
      },
    },
  ],
  "sortBy": Array [],
  "workspace": "testWorkspace",
}
`;

exports[`FunnelChart with  - default should create expected props for core chart 1`] = `
Object {
  "config": Object {},
}
`;

exports[`FunnelChart with arithmetic measures should create expected execution definition 1`] = `
Object {
  "attributes": Array [],
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "ah1EuQxwaCqs",
                },
              },
            },
            "localIdentifier": "m_ah1EuQxwaCqs",
          },
        },
        Object {
          "measure": Object {
            "alias": "Calculated 'Lost' measure",
            "definition": Object {
              "arithmeticMeasure": Object {
                "measureIdentifiers": Array [
                  "m_ah1EuQxwaCqs",
                  "m_afSEwRwdbMeQ",
                ],
                "operator": "difference",
              },
            },
            "localIdentifier": "m_m_ah1EuQxwaCqs_m_afSEwRwdbMeQ",
          },
        },
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "afSEwRwdbMeQ",
                },
              },
            },
            "localIdentifier": "m_afSEwRwdbMeQ",
          },
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [],
      "localIdentifier": "view",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [],
    },
    Object {
      "itemIdentifiers": Array [
        "measureGroup",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "ah1EuQxwaCqs",
            },
          },
        },
        "localIdentifier": "m_ah1EuQxwaCqs",
      },
    },
    Object {
      "measure": Object {
        "alias": "Calculated 'Lost' measure",
        "definition": Object {
          "arithmeticMeasure": Object {
            "measureIdentifiers": Array [
              "m_ah1EuQxwaCqs",
              "m_afSEwRwdbMeQ",
            ],
            "operator": "difference",
          },
        },
        "localIdentifier": "m_m_ah1EuQxwaCqs_m_afSEwRwdbMeQ",
      },
    },
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "afSEwRwdbMeQ",
            },
          },
        },
        "localIdentifier": "m_afSEwRwdbMeQ",
      },
    },
  ],
  "sortBy": Array [],
  "workspace": "testWorkspace",
}
`;

exports[`FunnelChart with arithmetic measures should create expected props for core chart 1`] = `Object {}`;

exports[`FunnelChart with assign color to attributes should create expected execution definition 1`] = `
Object {
  "attributes": Array [
    Object {
      "attribute": Object {
        "displayForm": Object {
          "identifier": "label.product.id.name",
        },
        "localIdentifier": "a_label.product.id.name",
      },
    },
  ],
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "ah1EuQxwaCqs",
                },
              },
            },
            "localIdentifier": "m_ah1EuQxwaCqs",
          },
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [
        Object {
          "attribute": Object {
            "displayForm": Object {
              "identifier": "label.product.id.name",
            },
            "localIdentifier": "a_label.product.id.name",
          },
        },
      ],
      "localIdentifier": "view",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [
        "measureGroup",
      ],
    },
    Object {
      "itemIdentifiers": Array [
        "a_label.product.id.name",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "ah1EuQxwaCqs",
            },
          },
        },
        "localIdentifier": "m_ah1EuQxwaCqs",
      },
    },
  ],
  "sortBy": Array [],
  "workspace": "testWorkspace",
}
`;

exports[`FunnelChart with assign color to attributes should create expected props for core chart 1`] = `
Object {
  "config": Object {
    "colorMapping": Array [
      Object {
        "color": Object {
          "type": "rgb",
          "value": Object {
            "b": 0,
            "g": 0,
            "r": 0,
          },
        },
        "predicate": [Function],
      },
      Object {
        "color": Object {
          "type": "rgb",
          "value": Object {
            "b": 0,
            "g": 0,
            "r": 255,
          },
        },
        "predicate": [Function],
      },
    ],
    "colorPalette": Array [
      Object {
        "fill": Object {
          "b": 73,
          "g": 49,
          "r": 195,
        },
        "guid": "01",
      },
      Object {
        "fill": Object {
          "b": 86,
          "g": 194,
          "r": 168,
        },
        "guid": "02",
      },
      Object {
        "fill": Object {
          "b": 177,
          "g": 217,
          "r": 243,
        },
        "guid": "03",
      },
      Object {
        "fill": Object {
          "b": 121,
          "g": 153,
          "r": 194,
        },
        "guid": "04",
      },
      Object {
        "fill": Object {
          "b": 34,
          "g": 37,
          "r": 162,
        },
        "guid": "05",
      },
    ],
  },
}
`;

exports[`FunnelChart with assign color to measures should create expected execution definition 1`] = `
Object {
  "attributes": Array [],
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "ah1EuQxwaCqs",
                },
              },
            },
            "localIdentifier": "m_ah1EuQxwaCqs",
          },
        },
        Object {
          "measure": Object {
            "alias": "Calculated 'Lost' measure",
            "definition": Object {
              "arithmeticMeasure": Object {
                "measureIdentifiers": Array [
                  "m_ah1EuQxwaCqs",
                  "m_afSEwRwdbMeQ",
                ],
                "operator": "difference",
              },
            },
            "localIdentifier": "m_m_ah1EuQxwaCqs_m_afSEwRwdbMeQ",
          },
        },
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "afSEwRwdbMeQ",
                },
              },
            },
            "localIdentifier": "m_afSEwRwdbMeQ",
          },
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [],
      "localIdentifier": "view",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [],
    },
    Object {
      "itemIdentifiers": Array [
        "measureGroup",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "ah1EuQxwaCqs",
            },
          },
        },
        "localIdentifier": "m_ah1EuQxwaCqs",
      },
    },
    Object {
      "measure": Object {
        "alias": "Calculated 'Lost' measure",
        "definition": Object {
          "arithmeticMeasure": Object {
            "measureIdentifiers": Array [
              "m_ah1EuQxwaCqs",
              "m_afSEwRwdbMeQ",
            ],
            "operator": "difference",
          },
        },
        "localIdentifier": "m_m_ah1EuQxwaCqs_m_afSEwRwdbMeQ",
      },
    },
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "afSEwRwdbMeQ",
            },
          },
        },
        "localIdentifier": "m_afSEwRwdbMeQ",
      },
    },
  ],
  "sortBy": Array [],
  "workspace": "testWorkspace",
}
`;

exports[`FunnelChart with assign color to measures should create expected props for core chart 1`] = `
Object {
  "config": Object {
    "colorMapping": Array [
      Object {
        "color": Object {
          "type": "rgb",
          "value": Object {
            "b": 0,
            "g": 0,
            "r": 0,
          },
        },
        "predicate": [Function],
      },
      Object {
        "color": Object {
          "type": "guid",
          "value": "05",
        },
        "predicate": [Function],
      },
    ],
    "colorPalette": Array [
      Object {
        "fill": Object {
          "b": 73,
          "g": 49,
          "r": 195,
        },
        "guid": "01",
      },
      Object {
        "fill": Object {
          "b": 86,
          "g": 194,
          "r": 168,
        },
        "guid": "02",
      },
      Object {
        "fill": Object {
          "b": 177,
          "g": 217,
          "r": 243,
        },
        "guid": "03",
      },
      Object {
        "fill": Object {
          "b": 121,
          "g": 153,
          "r": 194,
        },
        "guid": "04",
      },
      Object {
        "fill": Object {
          "b": 34,
          "g": 37,
          "r": 162,
        },
        "guid": "05",
      },
    ],
  },
}
`;

exports[`FunnelChart with single measure should create expected execution definition 1`] = `
Object {
  "attributes": Array [],
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "ah1EuQxwaCqs",
                },
              },
            },
            "localIdentifier": "m_ah1EuQxwaCqs",
          },
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [],
      "localIdentifier": "view",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [],
    },
    Object {
      "itemIdentifiers": Array [
        "measureGroup",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "ah1EuQxwaCqs",
            },
          },
        },
        "localIdentifier": "m_ah1EuQxwaCqs",
      },
    },
  ],
  "sortBy": Array [],
  "workspace": "testWorkspace",
}
`;

exports[`FunnelChart with single measure should create expected props for core chart 1`] = `Object {}`;

exports[`FunnelChart with single measure with viewBy should create expected execution definition 1`] = `
Object {
  "attributes": Array [
    Object {
      "attribute": Object {
        "displayForm": Object {
          "identifier": "label.product.id.name",
        },
        "localIdentifier": "a_label.product.id.name",
      },
    },
  ],
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "ah1EuQxwaCqs",
                },
              },
            },
            "localIdentifier": "m_ah1EuQxwaCqs",
          },
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [
        Object {
          "attribute": Object {
            "displayForm": Object {
              "identifier": "label.product.id.name",
            },
            "localIdentifier": "a_label.product.id.name",
          },
        },
      ],
      "localIdentifier": "view",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [
        "measureGroup",
      ],
    },
    Object {
      "itemIdentifiers": Array [
        "a_label.product.id.name",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "ah1EuQxwaCqs",
            },
          },
        },
        "localIdentifier": "m_ah1EuQxwaCqs",
      },
    },
  ],
  "sortBy": Array [],
  "workspace": "testWorkspace",
}
`;

exports[`FunnelChart with single measure with viewBy should create expected props for core chart 1`] = `Object {}`;

exports[`FunnelChart with single measure with viewBy sorted by attribute should create expected execution definition 1`] = `
Object {
  "attributes": Array [
    Object {
      "attribute": Object {
        "displayForm": Object {
          "identifier": "label.product.id.name",
        },
        "localIdentifier": "a_label.product.id.name",
      },
    },
  ],
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "ah1EuQxwaCqs",
                },
              },
            },
            "localIdentifier": "m_ah1EuQxwaCqs",
          },
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [
        Object {
          "attribute": Object {
            "displayForm": Object {
              "identifier": "label.product.id.name",
            },
            "localIdentifier": "a_label.product.id.name",
          },
        },
      ],
      "localIdentifier": "view",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [
        "measureGroup",
      ],
    },
    Object {
      "itemIdentifiers": Array [
        "a_label.product.id.name",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "ah1EuQxwaCqs",
            },
          },
        },
        "localIdentifier": "m_ah1EuQxwaCqs",
      },
    },
  ],
  "sortBy": Array [],
  "workspace": "testWorkspace",
}
`;

exports[`FunnelChart with single measure with viewBy sorted by attribute should create expected props for core chart 1`] = `Object {}`;

exports[`FunnelChart with two measures should create expected execution definition 1`] = `
Object {
  "attributes": Array [],
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "ah1EuQxwaCqs",
                },
              },
            },
            "localIdentifier": "m_ah1EuQxwaCqs",
          },
        },
        Object {
          "measure": Object {
            "definition": Object {
              "measureDefinition": Object {
                "item": Object {
                  "identifier": "afSEwRwdbMeQ",
                },
              },
            },
            "localIdentifier": "m_afSEwRwdbMeQ",
          },
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [],
      "localIdentifier": "view",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [],
    },
    Object {
      "itemIdentifiers": Array [
        "measureGroup",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "ah1EuQxwaCqs",
            },
          },
        },
        "localIdentifier": "m_ah1EuQxwaCqs",
      },
    },
    Object {
      "measure": Object {
        "definition": Object {
          "measureDefinition": Object {
            "item": Object {
              "identifier": "afSEwRwdbMeQ",
            },
          },
        },
        "localIdentifier": "m_afSEwRwdbMeQ",
      },
    },
  ],
  "sortBy": Array [],
  "workspace": "testWorkspace",
}
`;

exports[`FunnelChart with two measures should create expected props for core chart 1`] = `Object {}`;
